### DISCUSSION TITLE
Bug: Infinitely stuck

### DISCUSSION THREAD

**Discussion by GTorreil** [eslint/eslint#17422] [Q&A] [answered]
### Environment

Node version: v20.4.0
npm version: v9.7.2
Local ESLint version: v8.45.0 (Currently used)
Global ESLint version: v8.45.0
Operating System: darwin 22.2.0

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
require('@rushstack/eslint-patch/modern-module-resolution')

module.exports = {
  root: true,

  env: {
    browser: true,
    node: true,
  },

  extends: [
    'plugin:vue/vue3-recommended',
    'eslint:recommended',
    '@vue/typescript/recommended',
    '@vue/eslint-config-prettier',
  ],

  parserOptions: {
    ecmaVersion: 'latest',
  },

  rules: {
... Many rules !
}
```
</details>

- The project is a Ionic project. 
- I have the script : `"lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix --ignore-path .gitignore"` in my `package.json`. 
- Then I run `npm run lint` ...

### What did you expect to happen?

 I expect eslint to run and exit.

### What actually happened?

ESlint appears to be stuck. CPU Usage is 100% and the program never exits. I let it run for more than an hour. 
A colleague also on a mac has it running just fine. 

### Link to Minimal Reproducible Example

The problem is only on my machine

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I will be happy to help with additional data to troubleshoot this.

**Comment by Rec0iL99** [comment]
Hi @GTorreil, thanks for the issue. We would require a minimal example to fully understand why this is happening. Can you please provide one? Thanks.

**Comment by GTorreil** [comment]
@Rec0iL99 I'd be happy to provide you with this, but even my colleague on the same laptop model, and the same repo, can't reproduce it... So I only see two options : 
1. We remotely debug this (Just tell me what you want me to run, and i'll post the results)
2. We call it a mystery of CS and I reinstall my mac 😄

Cheers,

**Comment by nzakas** [comment]
@GTorreil If you can't reproduce on someone else's machine, then it's most likely related to your setup and not an issue with ESLint. If you can get a case where someone else can reproduce it, please post it here. We just don't have the bandwidth to hunt down mystery failures on individual computers.

Moving to a discussion.

**Comment by Ericlm** [comment]
I just wanted to report that I have the same issue, with an ionic app developed with vue & vite. The command runs this code : `"lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix --ignore-path .gitignore"`

**Comment by Ericlm** [comment, marked as answer]
I just fixed the issue! In fact, I was on an ionic project, and running `eslint .` would run on folders like `ios`, `android` and others, probably making eslint very slow. I just needed to change the path to `eslint ./src` to fix it 🙂

**Reply by KurtisIvey** [reply]
thank you! I just got a macbook pro m2 and couldn't figure this out for the life of me as I was on linux before. Such a simple solution, but now everything works

**Reply by artisticfox8** [reply]
You can discover folders it's spending a lot of time on with `eslint --debug . ` :)

And then add these folders to `ignores` list in `export default tseslint.config` in `eslint.config.js`

**Comment by alexboffey** [comment]
Adding to this as this thread seems to rank highly on Google.

Similar to @Ericlm above; eslint was hanging in my project as it was running through unnecessary folders.
In my case test assets `/playwright-report`, `/test-results` etc were missing from my `.eslintignore` file.

Thanks!

**Reply by artisticfox8** [reply]
Alternatively, you can also specify these in `eslint.config.js` on the `ignore` property list (as the part of the object passed to `export default tseslint.config`
